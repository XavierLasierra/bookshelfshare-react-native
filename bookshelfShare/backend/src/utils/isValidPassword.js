const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
