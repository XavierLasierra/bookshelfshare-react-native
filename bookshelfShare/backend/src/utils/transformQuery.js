function transformQuery(query) {
  let transformedQuery = {};
  const entries = Object.entries(query);
  if (entries.length > 0) {
    entries.forEach(([property, filter]) => {
      if (property.includes('_')) {
        const [propertyName, symbol] = property.split('_');
        transformedQuery = { ...transformedQuery, [propertyName]: { [`$${symbol}`]: filter } };
      } else {
        transformedQuery = { ...transformedQuery, [property]: { $regex: filter, $options: 'i' } };
      }
    });
  }
  return transformedQuery;
}

module.exports = { transformQuery };
