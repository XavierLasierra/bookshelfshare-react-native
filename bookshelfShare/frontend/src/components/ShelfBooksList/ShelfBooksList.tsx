import React from 'react';
import { FlatList, View } from 'react-native';
import ShelfBookElement from '../ShelfBookElement/ShelfBookElement';

export default function ShelfBooksList({
  shelfData, logo, navigation, shelfName
}: any) {
  function renderListElement({ item: { customInformation: { location }, bookData } }: any) {
    return (
      <ShelfBookElement
        navigation={navigation}
        logo={logo}
        bookData={bookData}
        location={location}
            // eslint-disable-next-line no-underscore-dangle
        key={bookData._id}
        shelfName={shelfName}
      />
    );
  }

  return (
    <>
      <FlatList
        showsVerticalScrollIndicator={false}
        data={shelfData}
        renderItem={renderListElement}
        keyExtractor={(item, index) => `shelf-${index}`}
      />
      <View style={{ height: 620 }} />
    </>
  );
}
