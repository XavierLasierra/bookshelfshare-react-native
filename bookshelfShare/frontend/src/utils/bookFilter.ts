export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
