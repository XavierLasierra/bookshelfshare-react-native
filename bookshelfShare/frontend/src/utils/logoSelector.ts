import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
