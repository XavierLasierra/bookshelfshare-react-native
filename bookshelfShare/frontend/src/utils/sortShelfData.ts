export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
