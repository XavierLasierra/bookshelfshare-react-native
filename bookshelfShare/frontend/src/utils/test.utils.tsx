import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
