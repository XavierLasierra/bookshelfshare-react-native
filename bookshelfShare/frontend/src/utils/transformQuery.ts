interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
