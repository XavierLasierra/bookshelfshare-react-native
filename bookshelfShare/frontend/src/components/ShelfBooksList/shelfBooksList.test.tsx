import React from 'react';
import ShelfBooksList from './ShelfBooksList';
import { render } from '../../utils/test.utils';
import shelfMock from '../../mocks/shelf.mock';

const navigation = {
  push: jest.fn()
};

describe('Given a ShelfBooksList component', () => {
  describe('When it is rendered', () => {
    test('Then should match the snapshot', () => {
      const screen = render(<ShelfBooksList
        shelfData={shelfMock.books}
        shelfName="My shelf"
        navigation={navigation}
      />);
      expect(screen).toMatchSnapshot();
    });
  });
});
