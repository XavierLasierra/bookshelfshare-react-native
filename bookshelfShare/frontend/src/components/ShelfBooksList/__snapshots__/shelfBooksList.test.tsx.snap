// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ShelfBookElement component When it is rendered Then should match the snapshot 1`] = `
Array [
  <RCTScrollView
    data={
      Array [
        Object {
          "bookData": Object {
            "authors": Array [
              "George Orwell",
            ],
            "categories": Array [
              "London (England)",
            ],
            "description": "Winston Smith is a worker at the Ministry of Truth, where he falsifies records for the party. Secretly subversive, he and his colleague Julia try to free themselves from political slavery but the price of freedom is betrayal.",
            "format": "BOOK",
            "images": Object {
              "smallThumbnail": "smallThumbnail1",
              "thumbnail": "thumbnail1",
            },
            "isbn": Object {
              "ISBN10": "0451524934",
              "ISBN13": "9780451524935",
            },
            "language": "en",
            "pageCount": 190,
            "publishedDate": "2009",
            "publisher": "No publisher found",
            "subtitle": "A Novel",
            "title": "1984",
          },
          "bookIsbn": "1",
          "customInformation": Object {
            "location": Array [
              0,
              0,
            ],
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginVertical": 10,
              "opacity": 1,
            }
          }
          testID="bookDetailButton"
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Row:
             
            1
             
            Column:
             
            1
             
            -
             
            My shelf
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "thumbnail1",
                }
              }
              style={
                Object {
                  "borderBottomLeftRadius": 10,
                  "borderTopLeftRadius": 10,
                  "height": 150,
                  "overflow": "hidden",
                  "width": 100,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomRightRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 150,
                  "justifyContent": "space-between",
                  "padding": 10,
                  "width": 210,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "width": 175,
                    }
                  }
                >
                  1984
                </Text>
                <Text>
                  George Orwell
                </Text>
                <Text>
                  No publisher found
                </Text>
              </View>
              <Text>
                ISBN13 -
                 
                9780451524935
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "height": 620,
      }
    }
  />,
]
`;
